text/3-entrees/off-chain-workers.md

```diff
@@ -0,0 +1,327 @@
+	}
+}
+
+// ...snip
+```
+
+There is a lot happening in the code. But basically we are:
+
+- Signing the `call` and `extra`, also called signed extension, and
+- Making the call(`call`, which includes the call paramters) and passing the sender `address`, signature of the data `signature`, and its signed extension `extra` on-chain as a transaction.
+
+We also define `SignedExtra` data type later in the runtime.
+
+src: [`runtimes/super-runtime/src/lib.rs`](https://github.com/substrate-developer-hub/recipes/tree/master/runtimes/super-runtime/src/lib.rs)
+
+```rust
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+	system::CheckVersion<Runtime>,
+	system::CheckGenesis<Runtime>,
+	system::CheckEra<Runtime>,
+	system::CheckNonce<Runtime>,
+	system::CheckWeight<Runtime>,
+	transaction_payment::ChargeTransactionPayment<Runtime>,
+);
+```
+
+### Sending Signed Transactions
+
+A signed transaction is sent with `T::SubmitSignedTransaction::submit_signed`, as shown below:
+
+src: [`pallets/offchain-demo/src/lib.rs`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/offchain-demo/src/lib.rs)
+
+```rust
+fn send_signed(block_number: T::BlockNumber) -> Result<(), Error<T>> {
+	use system::offchain::SubmitSignedTransaction;
+	//..snip
+
+	let submission: u64 = block_number.try_into().ok().unwrap() as u64;
+	let call = Call::submit_number_signed(submission);
+
+	// Using `SubmitSignedTransaction` associated type we create and submit a transaction
+	//   representing the call, we've just created.
+	let results = T::SubmitSignedTransaction::submit_signed(call);
+	for (acc, res) in &results {
+		match res {
+			Ok(()) => { debug::native::info!("off-chain send_signed: acc: {}| number: {}", acc, submission); },
+			Err(e) => {
+				debug::native::error!("[{:?}] Failed to submit signed tx: {:?}", acc, e);
+				return Err(<Error<T>>::SendSignedError);
+			}
+		};
+	}
+	Ok(())
+}
+```
+
+We have a function reference to `Call::submit_number_signed(submission)`. This is the function we are going to submit back to on-chain, and passing it to `T::SubmitSignedTransaction::submit_signed(call)`.
+
+You will notice that we run a for loop in the returned result. This implies that this call may make multiple transactions and return multiple results. It is because this call actually signs and sends the transaction with each of the accounts that can be found locally under the application crypto (which we defined earlier in `pub mod crypto {...}`). You can view this as the local accounts that are managed under this pallet namespace. Right now, we only have one key in the app crypto, so only one signed transaction is made.
+
+Eventually, the `call` transaction is made on-chain via the `create_transaction` function we defined earlier when we implemented `CreateTransaction` trait in our runtime.
+
+If you are wondering where we insert the local account in the pallet app crypto, it is actually in the node runtime.
```

Comment:
it is actually in the <del>node runtime</del> outer node's service. (link to rustdocs for substrate service)